lib/Eclair/LLVM/Codegen.hs

```diff
@@ -32,23 +30,23 @@ withLLVMTypeInfo ctx m = do
   structTys <- liftIO $ M.traverseWithKey (forwardDeclareStruct ctx) typedefs
 
   -- Then we serialize all types (including structs, with their bodies),
-  liftIO $ M.traverseWithKey (serialize ctx) structTys
+  _ <- liftIO $ M.traverseWithKey (serialize ctx) structTys
   -- Finally, we can call the function with all type info available in LLVM.
   m
   where
     forwardDeclareStruct ctx name structTy =
       (,structTy) <$> LibLLVM.mkOpaqueStructType ctx name
 
-    serialize :: ForeignPtr LibLLVM.Context -> Name -> (Ptr LibLLVM.Type, Type) -> IO ()
-    serialize ctx name (llvmTy, ty) = case ty of
+    serialize :: Foreign.ForeignPtr.ForeignPtr LibLLVM.Context -> Name -> (Ptr LibLLVM.Type, Type) -> IO ()
+    serialize ctx _ (llvmTy, ty) = case ty of
```

Comment:
Remove the `_` entirely, and change `M.traverseWithKey` on line 33 to `traverse`